service/telemetry_test.go

```diff
@@ -67,3 +87,133 @@ func TestBuildTelAttrs(t *testing.T) {
+			useOtel: false,
+			expectedMetrics: map[string]metricValue{
+				metricPrefix + ocPrefix + counterName: {
+					value: 13,
+					labels: map[string]string{
+						"service_name":        "otelcol",
+						"service_version":     "latest",
+						"service_instance_id": testInstanceID,
+					},
+				},
+			},
+		},
+		{
+			name:    "UseOpenTelemetryForInternalMetrics",
+			useOtel: true,
+			// TODO: add a test to verify that OTel is emitting a `target_info`
+			// 	info metric with resource attributes https://github.com/open-telemetry/opentelemetry-go/issues/3166
+			expectedMetrics: map[string]metricValue{
+				metricPrefix + ocPrefix + counterName: {
+					value: 13,
+					labels: map[string]string{
+						"service_name":        "otelcol",
+						"service_version":     "latest",
+						"service_instance_id": testInstanceID,
+					},
+				},
+				metricPrefix + otelPrefix + counterName: {
+					value:  13,
+					labels: map[string]string{},
+				},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			registry := featuregate.NewRegistry()
+			obsreportconfig.RegisterInternalMetricFeatureGate(registry)
+			require.NoError(t, registry.Apply(map[string]bool{obsreportconfig.UseOtelForInternalMetricsfeatureGateID: tc.useOtel}))
+
+			tel := newColTelemetry(registry)
+			buildInfo := component.NewDefaultBuildInfo()
+			cfg := telemetry.Config{
+				Resource: map[string]*string{
+					semconv.AttributeServiceInstanceID: &testInstanceID,
+				},
+			}
+
+			err := tel.initOnce(buildInfo, zap.NewNop(), cfg)
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, tel.shutdown())
+			}()
+
+			v := createTestMetrics(t, tel.mp)
+			defer func() {
+				view.Unregister(v)
+			}()
+
+			// Wait some time because OpenCensus can take sometime to set up and collect a measure.
+			time.Sleep(10 * time.Millisecond)
+
+			metrics := getMetricsFromPrometheus(t, tel.server.Handler)
+			require.Equal(t, len(tc.expectedMetrics), len(metrics))
+
+			for metricName, metricValue := range tc.expectedMetrics {
+				mf, present := metrics[metricName]
+				require.True(t, present, "expected metric %q was not present", metricName)
+				require.Len(t, mf.Metric, 1, "only one measure should exist for metric %q", metricName)
+
+				labels := make(map[string]string)
+				for _, pair := range mf.Metric[0].Label {
+					labels[pair.GetName()] = pair.GetValue()
+				}
+
+				require.Equal(t, metricValue.labels, labels, "labels for metric %q was different than expected", metricName)
+				require.Equal(t, metricValue.value, mf.Metric[0].Counter.GetValue(), "value for metric %q was different than expected", metricName)
+			}
+		})
+
+	}
+}
+
+func createTestMetrics(t *testing.T, mp metric.MeterProvider) *view.View {
+	// Creates a OTel Go counter
+	counter, err := mp.Meter("collector_test").SyncInt64().Counter(otelPrefix + counterName)
+	require.NoError(t, err)
+	counter.Add(context.Background(), 13)
+
+	// Creates a OpenCensus counter
```

Comment:
This is not an opencensus "counter". This is an opencensus measure + sum aggregation view :)